File: src/main/java/eu/codetopic/utils/ui/container/items/custom/EmptyCustomItem.java
/*
 * Copyright 2017 Jiří Kuchyňka (Anty)
 *
 *    Licensed under the Apache License, Version 2.0 (the "License");
 *    you may not use this file except in compliance with the License.
 *    You may obtain a copy of the License at
 *
 *        http://www.apache.org/licenses/LICENSE-2.0
 *
 *    Unless required by applicable law or agreed to in writing, software
 *    distributed under the License is distributed on an "AS IS" BASIS,
 *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 *    See the License for the specific language governing permissions and
 *    limitations under the License.
 */

package eu.codetopic.utils.ui.container.items.custom;

import android.content.Context;
import android.view.View;

import eu.codetopic.utils.R;

public class EmptyCustomItem extends CustomItem {

    @Override
    protected void onBindViewHolder(ViewHolder holder, int itemPosition) {
        holder.itemView.setVisibility(View.GONE);
    }

    @Override
    public int getItemLayoutResId(Context context) {
        return R.layout.item_empty;
    }
}

File: src/main/java/eu/codetopic/utils/ui/container/items/custom/EmptyCustomItem.kt
/*
 * utils
 * Copyright (C)   2017  anty
 *
 * This program is free  software: you can redistribute it and/or modify
 * it under the terms  of the GNU General Public License as published by
 * the Free Software  Foundation, either version 3 of the License, or
 * (at your option) any  later version.
 *
 * This program is distributed in the hope that it  will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied  warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.   See the
 * GNU General Public License for more details.
 *
 * You  should have received a copy of the GNU General Public License
 * along  with this program.  If not, see <http://www.gnu.org/licenses/>.
 */

package eu.codetopic.utils.ui.container.items.custom

import android.content.Context
import android.view.View

import eu.codetopic.utils.R

class EmptyCustomItem : CustomItem() {

    override fun onBindViewHolder(holder: CustomItem.ViewHolder, itemPosition: Int) {
        // Completely hide item, so there will nothing stay in layout
        holder.itemView.visibility = View.GONE
    }

    override fun getItemLayoutResId(context: Context): Int = R.layout.item_empty
}
